gossip/resolver.go

```diff
@@ -25,38 +25,62 @@ import (
 	"github.com/cockroachdb/cockroach/util/log"
 )
 
-// Resolver represents the different types of node resolvers.
-// Based on the ResolverType, it will return the same
+// Resolver is an interface which provides an abstract factory for
+// net.Addr addresses.
+type Resolver interface {
```

Comment:
I started that way, but it's just a lot of repetitive code. So I grouped those under the sockerResolver. The other resolver which makes sense is a urlResolver which could fetch the contents of a URL to get a list of hosts and then return those in random order until exhausted.